spring-integration-file/src/test/java/org/springframework/integration/file/tail/FileTailingMessageProducerTests.java

```diff
@@ -124,13 +131,48 @@ public void canRecalculateCommandWhenFileOrOptionsChanged() throws IOException {
 		adapter.stop();
 	}
 
+	@Test
+	@TailAvailable
+	public void testIdleEvent() throws Exception {
+		OSDelegatingFileTailingMessageProducer adapter = new OSDelegatingFileTailingMessageProducer();
+		adapter.setOptions(TAIL_OPTIONS_FOLLOW_NAME_ALL_LINES);
+		ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+		taskScheduler.afterPropertiesSet();
+		adapter.setTaskScheduler(taskScheduler);
+		CountDownLatch countDownLatch = new CountDownLatch(1);
+		adapter.setApplicationEventPublisher(event -> {
+			if (event instanceof FileTailingIdleEvent) {
+				FileTailingIdleEvent tailEvent = (FileTailingIdleEvent) event;
+				logger.debug(event);
+				countDownLatch.countDown();
+			}
+		});
+		File file = new File(testDir, "foo");
+		file.delete();
+		adapter.setFile(file);
+		QueueChannel outputChannel = new QueueChannel();
+		adapter.setOutputChannel(outputChannel);
+		adapter.setIdleEventInterval(100);
+		adapter.afterPropertiesSet();
+		adapter.start();
+		boolean noEvent = countDownLatch.await(500, TimeUnit.MILLISECONDS);
```

Comment:
M-m-m. Sorry, I'm very sensitive to an extra sleep for any our tests.
I would like to finish them as fast as possible 
Imagine like I run tests for the project at about one hundred times as day 😄 .

I guess there should be some special even when file doesn't exist. I'm pretty sure tailers should throw something on the matter.
So, it would be much better to catch that event as well via its particular `CountDownLatch`. And we have to ensure that there hasn't been any other events before. Like in our case we don't expect `FileTailingIdleEvent` if there is no file.

I don't say that your solution is wrong. Definitely it looks like the best compromise.
I just mean that the test would be better 😉 

That's all. 
Only what I need right now is a proper test-case.
Everything rest I will polish on merge and share here as a link to the commit.

So, I'm waiting from your for the last commit.